save number of eligible voters at end of initiative for later reference

tizip opened the pull request:
Added new model field "eligible_voters" to initiative.
Added migration script --> why is there another field change in this migration script?

Not sure about the default value (I chose 200).

Card: https://wekan.bewegung.jetzt/b/uQJxjrioxz2EG9nLi/initiativ-tool/kFLp7EtEb247iTHAe

joriki commented:
I think the reason for the other field change is that I corrected a typo in 27ed33046bec60e1f08684e10b6273b3b1f839be and didn't know that I should have created a migration to reflect it :-)

tizip commented:
> Why are you setting eligible_voters upon creation, and providing a default value?

I somehow thought we could make it easier for the frontend if we had a valid value there so it could be used for the display both during voting and after. But it doesn't really work. :/ I'll think of some other way to make it usable in the frontend (and remove the default value on creation again).

> In voty/initproc/management/commands/next_step.py, the calls to notify_followers() that are currently commented out would now occur before the call to save () 

Oh, good point, I hadn't thought of that.

Thanks! I'll fix both.

joriki commented:
> I somehow thought we could make it easier for the frontend if we had a valid value there so it could be used for the display both during voting and after.

How about if you create a method that returns the global current number of eligible voters before the vote ends and the instance's recorded number of eligible voters afterwards?

tizip commented:
OK, all review changes addressed and fixed. 

I added a cached property that returns all users while the ini has no value set, and the set value once it exists. The value is set by next_step once ini is closed.

I also added an (undesigned and consequently ugly) version to item.html. This still needs to be designed.

gnunicorn commented on voty/initproc/management/commands/next_step.py:
```diff
@@ -38,16 +38,19 @@ def handle(self, *args, **options):
                     try:
                         if i.is_accepted():
                             i.state = STATES.ACCEPTED
+                            i.eligible_voters = get_user_model().objects.count()
```
I glaube wir sollten hier ein `.filter(active=True)` mit rein nehmen.

otozyon replied:
Does it belong right at the end of `get_user_model().objects.count()`? Let me know or just append it and then I will merge this pull request.

tizip commented:
Included the changes requested by gnunicorn. I think it is now complete and can be merged.
